### PR TITLE
[CALCITE] Merge dialectic_parsing into master

### PR DESCRIPTION
**Pull request by michaelvdow**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java [hunk 1]
```diff
@@ -77,11 +77,33 @@
...
+
+  /** Conformance value that instructs Calcite to use SQL semantics
+   * consistent with Redshift. */
+  REDSHIFT,
+
+  /** Conformance value that instructs Calcite to use SQL semantics
+   * consistent with MySQL version 8.x. */
+  MYSQL_8;
+
+  // TODO: change to isLeftSemiJoinAllowed, only place it's used. Remove BABEL.
```
Comment (AndrewPochapsky) [inline core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:102]: nit: Are we going to deal with this TODO?

Comment (michaelvdow) [reply core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:102]: I forgot about that, did we want to remove the BABEL SqlConformanceEnum?

Comment (dasch-1) [reply core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:102]: Yes, let's remove it.

Comment (dasch-1) [reply core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:102]: In fact, we should double check that there are no references to "server" or "babel" throughout the entire codebase.

Comment (michaelvdow) [reply core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:102]: In places like [this](https://github.com/googleinterns/calcite/blob/master/core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java#L481), should we use like the LENIENT conformance enum?

Comment (dasch-1) [reply core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:102]: Yeah, let's use LENIENT.

**Pull request merged**